src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strArray = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (this[i] === undefined || this[i] === null) {
+        element = '';
+      }
+
+      strArray += element;
```

Comment:
actually, adding an empty string does nothing, so change the condition to `this[i] !== undefined || this[i] !== null` and move strArr += element to `if` block